src/core/gltf2exporter/dracocompressor_p.cpp

```diff
@@ -0,0 +1,312 @@
+#include <fstream>
+#include <iterator>
+#include <iostream>
+
+namespace Kuesa::Compression {
+namespace {
+draco::GeometryAttribute::Type attributeTypeFromName(const QString &attributeName)
+{
+    auto type = draco::GeometryAttribute::Type::INVALID;
+
+    if (attributeName == Qt3DRender::QAttribute::defaultPositionAttributeName())
+        type = draco::GeometryAttribute::Type::POSITION;
+    if (attributeName == Qt3DRender::QAttribute::defaultNormalAttributeName())
+        type = draco::GeometryAttribute::Type::NORMAL;
+    if (attributeName == Qt3DRender::QAttribute::defaultColorAttributeName())
+        type = draco::GeometryAttribute::Type::COLOR;
+    if (attributeName == Qt3DRender::QAttribute::defaultTextureCoordinateAttributeName()
+     || attributeName == Qt3DRender::QAttribute::defaultTextureCoordinate1AttributeName()
+     || attributeName == Qt3DRender::QAttribute::defaultTextureCoordinate2AttributeName())
+        type = draco::GeometryAttribute::Type::TEX_COORD;
+    if (attributeName == Qt3DRender::QAttribute::defaultTangentAttributeName())
+        type = draco::GeometryAttribute::Type::GENERIC;
+    if (attributeName == Qt3DRender::QAttribute::defaultJointWeightsAttributeName())
+        type = draco::GeometryAttribute::Type::GENERIC;
+    if (attributeName == Qt3DRender::QAttribute::defaultJointIndicesAttributeName())
+        type = draco::GeometryAttribute::Type::GENERIC;
+
+    return type;
+}
+
+template<typename T>
+struct attribute_type_trait {
+    static constexpr draco::DataType type = draco::DataType::DT_INVALID;
+};
+
+template<>
+struct attribute_type_trait<float> {
+    static constexpr draco::DataType type = draco::DataType::DT_FLOAT32;
+};
+
+template<>
+struct attribute_type_trait<char> {
+    static constexpr draco::DataType type = draco::DataType::DT_INT8;
+};
+
+template<>
+struct attribute_type_trait<unsigned char> {
+    static constexpr draco::DataType type = draco::DataType::DT_UINT8;
+};
+
+template<>
+struct attribute_type_trait<unsigned short> {
+    static constexpr draco::DataType type = draco::DataType::DT_UINT16;
+};
+
+template<>
+struct attribute_type_trait<short> {
+    static constexpr draco::DataType type = draco::DataType::DT_INT16;
+};
+
+template<>
+struct attribute_type_trait<int> {
+    static constexpr draco::DataType type = draco::DataType::DT_INT32;
+};
+
+template<>
+struct attribute_type_trait<unsigned int> {
+    static constexpr draco::DataType type = draco::DataType::DT_UINT32;
+};
+
+template<>
+struct attribute_type_trait<double> {
+    static constexpr draco::DataType type = draco::DataType::DT_FLOAT64;
+};
+
+template<typename T>
+QByteArray packedDataInBuffer(const QByteArray &inputBuffer,
+                              int vertexSize,
+                              int offset,
+                              int stride,
+                              int count)
+{
+    QByteArray outputBuffer;
+    outputBuffer.resize(count * vertexSize * sizeof(T));
```

Comment:
vertexSize * sizeof(T) could probably cached into a variable since it's reused in several places

Reply:
done